auth.go

```diff
@@ -387,36 +402,49 @@ func (x *Central) GetTokenFromIdentityPassword(identity, password string) (*Toke
 // Create a new session. Returns a session key, which can be used in future to retrieve the token.
 // The internal session expiry is controlled with the member NewSessionExpiresAfter.
 // The session key is typically sent to the client as a cookie.
-func (x *Central) Login(identity, password string) (sessionkey string, token *Token, e error) {
-	token = &Token{}
-	if token.Identity, e = x.authenticator.Authenticate(identity, password); e == nil {
-		x.Log.Infof("Login authentication success (%v)", identity)
-		var permit *Permit
-		if permit, e = x.permitDB.GetPermit(identity); e == nil {
-			if x.MaxActiveSessions != 0 {
-				if e = x.sessionDB.InvalidateSessionsForIdentity(identity); e != nil {
-					x.Log.Errorf("Invalidate sessions for identity (%v) failed when enforcing MaxActiveSessions (%v)", identity, e)
-					return "", nil, e
-				}
-			}
-			token.Expires = time.Now().Add(x.NewSessionExpiresAfter)
-			token.Permit = *permit
-			sessionkey = generateSessionKey()
-			if e = x.sessionDB.Write(sessionkey, token); e == nil {
-				x.Stats.IncrementGoodLogin(x.Log)
-				x.Log.Infof("Login successful (%v)", identity)
-				return
-			}
-		} else {
-			x.Log.Infof("Login GetPermit failed (%v) (%v)", identity, e)
-		}
-	} else {
+func (x *Central) Login(identity, password string) (string, *Token, error) {
+	token := &Token{}
+	if err := x.authenticator.Authenticate(identity, password); err != nil {
 		x.Stats.IncrementInvalidPasswords(x.Log)
-		x.Log.Infof("Login Authentication failed (%v) (%v)", identity, e)
+		x.Log.Infof("Login Authentication failed (%v) (%v)", identity, err)
+		return "", nil, err
+	}
+
+	var userId UserId
```

Comment:
It's customary for Go code such as this to be written in the following manner:
```go
userId, err := x.userStore.GetUserIdFromIdentity(identity)
if err != nil {
// ...
```
It is shorter this way.